palace/linalg/vector.hpp

```diff
@@ -0,0 +1,191 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+#ifndef PALACE_LINALG_VECTOR_HPP
+#define PALACE_LINALG_VECTOR_HPP
+
+#include <complex>
+#include <mfem.hpp>
+#include "utils/communication.hpp"
+
+namespace palace
+{
+
+using Operator = mfem::Operator;
+using Vector = mfem::Vector;
+
+//
+// Functionality extending mfem::Vector from MFEM, including basic functions for parallel
+// vectors distributed across MPI processes.
+//
+
+// A complex-valued vector represented as two real vectors, one for each component.
+class ComplexVector
+{
+private:
+  Vector x, xr, xi;
+
+public:
+  // Create a vector with the given size.
+  ComplexVector(int n = 0);
+
+  // Copy constructor.
+  ComplexVector(const ComplexVector &y);
```

Comment:
As discussed offline, I've added some subset of the functionality suggested in `hughcars/driven-port-debug-comments`. For now I've skipped the move constructor since we haven't had a need for it.